flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergSink.java

```diff
@@ -545,6 +561,34 @@ public DataStreamSink<RowData> append() {
       }
       return rowDataDataStreamSink;
     }
+
+    @VisibleForTesting
+    List<Integer> checkAndGetEqualityFieldIds() {
```

Comment:
This is a copy-paste from `FlinkSink`. It is used in `TestFlinkIcebergSinkV2`. I see that this test behaves differently for `IcebergSink` for `WRITE_DISTRIBUTION_MODE_RANGE` and turned off partitioning: `Equality field columns shouldn't be empty when configuring to use UPSERT data stream` is thrown, instead of `Invalid write distribution mode ...`. Do you have an idea whether it is correct or unexpected behaviour?

Reply:
`checkAndGetEqualityFieldIds()` was moved to `SinkUtil` class. This method is already being referenced by both Sinks implementations.  
Looks Like I forgot to delete this method from `FlinkSink`. So we can do the following
1. Don't add this method to `IcebergSink`
2. remove the method from `FlinkSink`
3. the test `TestFlinkIcebergSinkV2::testCheckAndGetEqualityFieldIds` can be rewritten as:
```java
  @TestTemplate
  public void testCheckAndGetEqualityFieldIds() {
    table
        .updateSchema()
        .allowIncompatibleChanges()
        .addRequiredColumn("type", Types.StringType.get())
        .setIdentifierFields("type")
        .commit();

    DataStream<Row> dataStream =
        env.addSource(new BoundedTestSource<>(ImmutableList.of()), ROW_TYPE_INFO);

    // Use schema identifier field IDs as equality field id list by default
    assertThat(SinkUtil.checkAndGetEqualityFieldIds(table, null))
        .containsExactlyInAnyOrderElementsOf(table.schema().identifierFieldIds());

    // Use user-provided equality field column as equality field id list
    assertThat(SinkUtil.checkAndGetEqualityFieldIds(table, Lists.newArrayList("id")))
        .containsExactlyInAnyOrder(table.schema().findField("id").fieldId());

    assertThat(SinkUtil.checkAndGetEqualityFieldIds(table, Lists.newArrayList("type")))
        .containsExactlyInAnyOrder(table.schema().findField("type").fieldId());
  }
```

This should work but please let me know if it doesn't. Thanks!